ballerina/vector_store.bal

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/ai;
+import ballerinax/milvus;
+
+# Milvus Vector Store implementation with support for Dense, Sparse, and Hybrid vector search modes.
+#
+# This class implements the ai:VectorStore interface and integrates with the Milvus vector database
+# to provide functionality for vector upsert, query, and deletion.
+#
+public isolated class VectorStore {
+    *ai:VectorStore;
+
+    private final milvus:Client milvusClient;
+    private final Configuration config;
+    private final string chunkFieldName;
+    private final int topK;
+
+    # Initializes the Milvus vector store with the given configuration.
+    #
+    public isolated function init(
+            @display {label: "Service URL"} string serviceUrl,
+            @display {label: "Milvus Configuration"} Configuration config,
```

Comment:
Do we need two fields for this, cant we do it in a one parameter?

Reply:
Decided to proceed with the current approach since `ai.pinecone` already uses this design. Thought of maintaining standard parameters across all vector stores for consistency.